Add interface identifier conversion

faern opened the pull request:
Adding bindings to functions allowing converting from network interface aliases, names, GUIDs and indexes to their `NET_LUID` representation, and back again.

faern committed:
Add network interface conversion functions

* Add ifdef module and NET_LUID definition

* Define NET_IFINDEX, NET_IFTYPE

* Add to ConvertInterfaceXtoY for Luid, Alias, Name, Index, Guid

* Add tests for struct layout of NET_LUID

retep998 commented on src/shared/netioapi.rs:
```diff
@@ -0,0 +1,49 @@
+// except according to those terms.
+use shared::basetsd::SIZE_T;
+use shared::guiddef::GUID;
+use shared::ifdef::{NET_IFINDEX,NET_LUID,PNET_IFINDEX,PNET_LUID};
```
There should be a space after the comma.

faern replied:
Fixed

retep998 commented on src/shared/netioapi.rs:
```diff
@@ -0,0 +1,49 @@
+use shared::guiddef::GUID;
+use shared::ifdef::{NET_IFINDEX,NET_LUID,PNET_IFINDEX,PNET_LUID};
+use shared::minwindef::DWORD;
+use shared::ntdef::{PWSTR,WCHAR};
```
Here as well.

faern replied:
Fixed

retep998 commented on src/shared/ifdef.rs:
```diff
@@ -0,0 +1,23 @@
+// Copyright © 2015-2017 winapi-rs developers
```
Shouldn't the copyright year be just 2018?

faern replied:
Fixed

retep998 commented on src/shared/netioapi.rs:
```diff
@@ -0,0 +1,49 @@
+// Copyright © 2015-2017 winapi-rs developers
```
Here as well.

faern replied:
Fixed

faern commented:
I added the `A` version where there were `W` versions present. Also squashed everything to one commit and added the PR number to the commit message, as seems to be the standard here.

retep998 commented:
The adding of the PR number to the commit message is done by the Squash and merge feature of Github which I use to merge PRs.

faern commented:
All feedback has been addressed. So I think this is ready for merge?
